qt/widgets/common/inc/MantidQtWidgets/Common/ScriptEditor.h

```diff
@@ -85,6 +85,8 @@ class EXPORT_OPT_MANTIDQT_COMMON ScriptEditor : public QsciScintilla {
   /// Capture key presses
   void keyPressEvent(QKeyEvent *event) override;
   /// The current filename
+  // this needs to return as value as it is an inline function
+  // cppcheck-suppress returnByReference
   inline QString fileName() const { return m_filename; }
```

Comment:
```suggestion
  const inline QString &fileName() const { return m_filename; }
```
With `const` the `inline` works with returning by reference, I just gave this a little test